allennlp/data/dataset_readers/reading_comprehension/triviaqa.py

```diff
@@ -0,0 +1,162 @@
+            question_text = question_json['Question']
+            question_tokens = self._tokenizer.tokenize(question_text)
+
+            evidence_files: List[List[str]] = []  # contains lines from each evidence file
+            if 'web' in file_path:
+                for result in question_json['SearchResults']:
+                    filename = result['Filename']
+                    evidence_file = base_tarball.extractfile(os.path.join("evidence", "web", filename))
+                    evidence_files.append([line.decode('utf-8') for line in evidence_file.readlines()])
+            else:
+                for result in question_json['EntityPages']:
+                    filename = result['Filename']
+                    evidence_file = base_tarball.extractfile(os.path.join("evidence", "wikipedia", filename))
+                    evidence_files.append([line.decode('utf-8') for line in evidence_file.readlines()])
+
+            answer_json = question_json['Answer']
+            answer_texts = answer_json['NormalizedAliases'] + answer_json.get('HumanAnswers', [])
+            for paragraph in self.pick_paragraphs(evidence_files, question_text, answer_texts):
+                instance = self.text_to_instance(question_text, paragraph, answer_texts, question_tokens)
+                if instance:
+                    instances.append(instance)
+        if not instances:
+            raise ConfigurationError("No instances were read from the given filepath {}. "
+                                     "Is the path correct?".format(file_path))
+        return Dataset(instances)
+
+    def pick_paragraphs(self,
+                        evidence_files: List[List[str]],
+                        question: str = None,
+                        answer_texts: List[str] = None) -> List[str]:
+        """
+        Given a list of evidence documents, return a list of paragraphs to use as training
+        examples.  Each paragraph returned will be made into one training example.
+
+        To aid in picking the best paragraph, you can also optionally pass the question text or the
+        answer strings.  Note, though, that if you actually use the answer strings for picking the
+        paragraph on the dev or test sets, that's likely cheating, depending on how you've defined
+        the task.
+        """
+        # pylint: disable=no-self-use,unused-argument
+        paragraphs = []
+        for evidence_file in evidence_files:
+            whole_document = ' '.join(evidence_file)
+            tokens = whole_document.split(' ')
+            paragraph = ' '.join(tokens[:400])
+            paragraphs.append(paragraph)
+        return paragraphs
+
+    @overrides
+    def text_to_instance(self,  # type: ignore
+                         question_text: str,
+                         passage_text: str,
+                         answer_texts: List[str] = None,
+                         question_tokens: List[Token] = None) -> Instance:
```

Comment:
This type annotation should be `Optional[Instance]`.

Reply:
Thanks, good point.

Reply:
This made me realize that making this optional was actually a bad decision - I moved the skipping logic to outside of this function, so this always returns an `Instance`.